documentation/api/addAssertion.md

```diff
@@ -10,82 +10,208 @@ expect.addAssertion(pattern, handler);
 expect.addAssertion([pattern, ...]], handler);
 ```
 
-New assertions can be added to Unexpected the following way.
+`expect.addAssertion` takes two arguments:
+
+1. a string pattern (or an array of patterns) that describes the assertion.
+2. a handler function that is called when when the assertion is invoked.
+
+For example:
 
 ```js
-var errorMode = 'default'; // use to control the error mode later in the example
 expect.addAssertion(
-  '<array> [not] to be (sorted|ordered) <function?>',
-  function(expect, subject, cmp) {
-    expect.errorMode = errorMode;
-    expect(subject, '[not] to equal', [].concat(subject).sort(cmp));
+  '<array> to have item <any>',
```

Comment:
Maybe we should make something a bit more fun. Like `to have sum`:

```js
expect([0,1,2,3,4], 'to have sum', 10)
```

Reply:
Okay I can see you have lots of stuff pointing back to this, so let's just keep it the way it is.

Reply:
Yeah I spent way too much time trying to come up with an assertion I could build up on. I can make it work with `to have sum`, do you have an idea how we could add alternations to it?

Reply:
One idea: you could allow floating poing numbers and then have "floored" as an alternation that causes `Math.floor()` on the result before comparison?

Reply:
I guess that would be a flag? Anyway, too much work :grin: